lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -3617,6 +3628,17 @@ insertTensorComputationStartEndTerminate(ArrayRef<SILValue> resultValues)
   };
 }
 
+/// Returns true when this function must be entirely lowered to a TF graph
+/// function, with no host-side logic remaining (i.e., no sends/recvs, and no
+/// start/stop tensor computation on the host side). In other words, this
+/// function uses the tensorflow calling convention.
+///
+/// The only way to call/use such a function is from a TF graph node (e.g. by
+/// referencing the function in a function-typed op attribute).
+static bool isAcceleratorOnly(const SILFunction &hostFn) {
+  // TODO: replace the impl with a proper function-level attribute.
+  return hostFn.getName().contains("__magic_tf_only__");
```

Comment:
You can simply check 
hostFn.getRepresentation() == SILFunctionType::Representation::TensorFlow. 

My patch handles the part from SILGen to diagnosing after the marking pass. Will send it out today.

Reply:
Thanks for the suggestion. Extended the TODO comment with it -- I'll defer the code change to a later PR, so that I can understand more about the tensorflow convention vs the new attr added in https://github.com/apple/swift/pull/17460 first.